### DISCUSSION TITLE
mitmproxy on Apple silicon

### DISCUSSION THREAD

**Discussion by keeely** [mitmproxy/mitmproxy#6507] [General]
Sorry if this is a silly question, but is mitmproxy written in Python?  pip installing on Apple silicon gives me an intel binary at /opt/homebrew/bin/mitmproxy, which presumably won't run unless I install Rosetta.  Is this a bug? I'd expect it to just refuse to install or something.  It's a similar story installing the Homebrew cask, it gives me an intel binary, even when using the ARM Homebrew.

**Comment by mhils** [comment]
`pip install` should give you an arm64 version. The `mitmproxy` executable created by pip should be a regular `console_scripts` entrypoint / built based on https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#creating-executable-scripts. Are you sure the binary in homebrew/bin does not stem from a previous cask install?

The Homebrew cask is x86_64 only at the moment because GitHub Actions does not support M1 builds for FOSS yet and we don't want to overcomplicate our CI infra. See https://github.com/Homebrew/homebrew-core/pull/145547 for details on why we needed to move from a formula to a cask. Once GitHub Actions has a proper M1 story, we'll provide arm64 casks as well. :)

**Comment by keeely** [comment]
My apologies, I tried this again on a fresh Parallels ARM virtual machine using the built-in Python and the pip install gives me the usual text mode console script stub, so it must have been the left-over cask.  I guess that's the problem with pip3 putting binaries in the same place as the cask.  Looking forward to using the ARM cask instead!
